### DISCUSSION TITLE
Closing existing Agroal connection and getting a new connection when existing configuration parameters have expired

### DISCUSSION THREAD

**Discussion by supre** [quarkusio/quarkus#38765] [Community]
At my company, we have a project where we use Agroal for managing the connection pool. We have a multitenant architecture where the credentials for each tenant are stored in another vault. 
Occasionally, we might move the database to a new server, and when we do that the existing connection to the tenant in the connection pool won't work. The issue we are facing is how to close the existing connection to a tenant when the existing connection has become obsolete and create a new connection using the updated parameters in the vault.  
What could be the cleanest way to close the existing connection and recreate the connection using the updated parameters in the vault?

**Comment by barreiro** [comment]
In the Agroal library, the `AgroalDatasource` object has a [`flush()`](https://github.com/agroal/agroal/blob/2.3/agroal-api/src/main/java/io/agroal/api/AgroalDataSource.java#L72) method that could be used to evict and repopulate the pool.

As far as I know, at the moment there is no mechanism in Quarkus to flush the pool but if you can somehow cast/unwrap a `Datasource` to `AgroalDatasource` you should be able to implement something custom.
